tests/test_writing.py

```diff
@@ -72,7 +72,9 @@ def validate_catalog_link_type(
             href: str, link_type: str, should_include_self: bool
         ) -> None:
             cat_dict = pystac.StacIO.default().read_json(href)
-            cat = pystac.Catalog.from_file(href)
+            cat = pystac.read_file(href)
+            if not isinstance(cat, pystac.Catalog):
+                raise pystac.STACTypeError(f"File at {href} is not a Catalog.")
```

Comment:
@lossyrob @matthewhanson One consequence of adding the `STACObject.identify` methods is that `Catalog.from_file` will now fail on a Collection file, which is why these lines changed. Judging from discussion in radiantearth/stac-spec#1082 I think this is the right behavior, but let me know if you feel otherwise. If we feel it should succeed, I can update `Catalog.identify` accordingly.

Reply:
I think that's the right behavior. A Collection is not a Catalog, so should not be able to be opened up as such